packages/react-components/src/components/RichTextEditor/RichTextSendBox.tsx

```diff
@@ -269,22 +345,47 @@ export const RichTextSendBox = (props: RichTextSendBoxProps): JSX.Element => {
     setAttachmentUploadsPendingError(undefined);
 
     /* @conditional-compile-remove(file-sharing-acs) */
-    if (hasIncompleteAttachmentUploads(attachments)) {
+    /* @conditional-compile-remove(rich-text-editor-image-upload) */
+    if (
+      /* @conditional-compile-remove(file-sharing-acs) */ hasIncompleteAttachmentUploads(attachments) ||
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */ hasIncompleteAttachmentUploads(
+        imageUploadsInProgress
+      )
+    ) {
       setAttachmentUploadsPendingError({ message: strings.attachmentUploadsPendingError, timestamp: Date.now() });
       return;
     }
 
-    const message = contentValue;
+    let message = contentValue;
 
     // we don't want to send empty messages including spaces, newlines, tabs
     // Message can be empty if there is a valid attachment upload
-    if (hasContent || /* @conditional-compile-remove(file-sharing-acs) */ isAttachmentUploadCompleted(attachments)) {
+    if (
+      hasContent ||
+      /* @conditional-compile-remove(file-sharing-acs) */ isAttachmentUploadCompleted(attachments) ||
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */ isAttachmentUploadCompleted(
+        imageUploadsInProgress
+      )
+    ) {
+      /* @conditional-compile-remove(file-sharing-acs) */
+      let attachmentArray = toAttachmentMetadata(attachments);
+
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */
+      if (imageUploadsInProgress) {
```

Comment:
Should we add check for `imageArray` length here? (So it will be be checked before 
`message = addUploadedImagesToMessage(message, imageUploadsInProgress);`)